### ISSUE TITLE
AWSMachinePool not reconciled after move

### ISSUE THREAD

**Issue by dkoshkin** [kubernetes-sigs/cluster-api-provider-aws#2393]
/kind bug

**What steps did you take and what happened:**
[A clear and concise description of what the bug is.]

1. Create a cluster with an `AWSMachinePool`
2. `clusterctl move` to the new cluster
3. `awscluster` and `awsmachineTemplate` reconcile as expected but `awsmachinepools` do not reconcile with error:

```
Status:
  Conditions:
    Last Transition Time:  2021-05-10T13:05:03Z
    Message:               InvalidLaunchTemplateName.AlreadyExistsException: Launch template name already in use.
                           status code: 400, request id: 7724f84c-7285-4466-b5cd-2ab49381c45f
    Reason:                LaunchTemplateCreateFailed
    Severity:              Error
    Status:                False
    Type:                  Ready
    Last Transition Time:  2021-05-10T12:01:28Z
    Reason:                WaitingForClusterInfrastructure
    Severity:              Info
    Status:                False
    Type:                  ASGReady
    Last Transition Time:  2021-05-10T13:05:03Z
    Message:               InvalidLaunchTemplateName.AlreadyExistsException: Launch template name already in use.
                           status code: 400, request id: 7724f84c-7285-4466-b5cd-2ab49381c45f
    Reason:                LaunchTemplateCreateFailed
    Severity:              Error
    Status:                False
    Type:                  LaunchTemplateReady
```

**What did you expect to happen:**
AWSMachinePool should reconcile after `clusterctl move`.

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

I've tracked down where this error is occurring:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/c7c8ae1ee071afbb33c3b45b0f9adbea149581ce/exp/controllers/awsmachinepool_controller.go#L390-L403

This is calling
```
ec2svc.GetLaunchTemplate(machinePoolScope.AWSMachinePool.Status.LaunchTemplateID)
```
However, after `clusterctl move` the value for `machinePoolScope.AWSMachinePool.Status.LaunchTemplateID` is empty. This makes sense since its the reconcilers job to set the `status` but somehow an existing Launch Template needs to be discovered.

Perhaps we need to bring back https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/2184

**Environment:**

- Cluster-api-provider-aws version: https://github.com/kubernetes-sigs/cluster-api-provider-aws/releases/tag/v0.6.5
- Kubernetes version: (use `kubectl version`): `1.20.6`
- OS (e.g. from `/etc/os-release`):

**Comment by sedefsavas** [comment]
I am able to replicate the problem and have a possible fix.

/assign
/kind backport
/milestone v0.7.0

**Comment by sedefsavas** [comment]
Fixed by #2394
/close
